src/test/java/htsjdk/tribble/bed/BEDCodecTest.java

```diff
@@ -25,27 +25,63 @@
 package htsjdk.tribble.bed;
 
 import htsjdk.HtsjdkTest;
+import htsjdk.samtools.util.BlockCompressedFilePointerUtil;
+import htsjdk.samtools.util.BlockCompressedInputStream;
 import htsjdk.tribble.AbstractFeatureReader;
 import htsjdk.tribble.Feature;
+import htsjdk.tribble.FeatureReader;
 import htsjdk.tribble.TestUtils;
 import htsjdk.tribble.annotation.Strand;
 import htsjdk.tribble.bed.FullBEDFeature.Exon;
-import htsjdk.tribble.index.IndexFactory;
-import htsjdk.tribble.index.linear.LinearIndex;
 import htsjdk.tribble.index.tabix.TabixFormat;
-import htsjdk.tribble.util.LittleEndianOutputStream;
+import htsjdk.tribble.readers.AsciiLineReaderIterator;
+import htsjdk.tribble.readers.LineIterator;
+import htsjdk.tribble.readers.PositionalBufferedStream;
+import htsjdk.tribble.util.ParsingUtils;
 import org.testng.Assert;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
 import java.awt.*;
-import java.io.BufferedOutputStream;
 import java.io.File;
-import java.io.FileOutputStream;
 import java.io.IOException;
+import java.io.InputStream;
 import java.util.List;
 
 public class BEDCodecTest extends HtsjdkTest {
 
+    @DataProvider(name = "gzippedBedTestData")
+    public Object[][] getBedTestData(){
+        return new Object[][] {
+                {
+                    // BGZP BED file with no header, 2 features
+                    new File(TestUtils.DATA_DIR, "bed/2featuresNoHeader.bed.gz"), 0  // header has length 0
+                },
+                {
+                    // BGZP BED file with one line header, 2 features
+                    new File(TestUtils.DATA_DIR, "bed/2featuresWithHeader.bed.gz"), 10 // header has length 10
+
+                }
+        };
+    }
+
+    @Test(dataProvider = "gzippedBedTestData")
+    public void testReadActualHeader(final File gzippedBedFile, final int firstFeatureOffset) throws IOException {
+        // Given an indexable SOURCE on a BED file, test that readActualHeader retains the correct offset
+        // of the first feature, whether there is a header or not
+        BEDCodec bedCodec = new BEDCodec();
+        try (final InputStream is = ParsingUtils.openInputStream(gzippedBedFile.getPath());
+             final BlockCompressedInputStream bcis = new BlockCompressedInputStream(is))
+        {
```

Comment:
A matter of taste :)
One data point, oracle's Java documentation: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html